flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -551,7 +551,7 @@ void onSenderBacklog(int backlog) throws IOException {
 	}
 
 	public void onBuffer(Buffer buffer, int sequenceNumber, int backlog) throws IOException {
-		boolean recycleBuffer = true;
+		boolean enqueued = false;
```

Comment:
> In case of some parse exception, the whole TM is restarted anyways.

That's true for now, but it can change in the future.

> renaming and inversion is unnecessary

This is to give more context to prevent future bugs.

Reply:
While `enqueued` is more semantic, I'd assume that the technical `recycleBuffer` should be more explicit concerning bugs.

Reply:
Reverted to minimize changes in this PR.